src/lib/reasoners/adt_rel.ml

```diff
@@ -431,6 +431,8 @@ let assume_literals la uf env =
        match lit with
        | Eq (r1, r2) as l, _, ex, Th_util.Subst when is_adt r1 ->
          Debug.assume l;
+         let r1, _ = Uf.find_r uf r1 in
```

Comment:
This one looks wrong?

Reply:
Ok, I'm not sure to understand what's going on here. I didn't find a failure-induced input.

Reply:
To clarify; nothing needs to be done *for `Subst` equalities specifically*.